crates/bevy_render/src/view/mod.rs

```diff
@@ -129,40 +169,209 @@ impl ExtractedView {
+    /// [`ColorGradingGlobal::midtones_range`] field.
+    pub midtones: ColorGradingSection,
+
+    /// Color grading values that are applied to the lighter parts of the image.
+    ///
+    /// The cutoff points can be customized with the
+    /// [`ColorGradingGlobal::midtones_range`] field.
+    pub highlights: ColorGradingSection,
+}
+
+/// Filmic color grading values applied to the image as a whole (as opposed to
+/// individual sections, like shadows and highlights).
+#[derive(Clone, Debug, Reflect)]
+#[reflect(Default)]
+pub struct ColorGradingGlobal {
     /// Exposure value (EV) offset, measured in stops.
     pub exposure: f32,
 
-    /// Non-linear luminance adjustment applied before tonemapping. y = pow(x, gamma)
-    pub gamma: f32,
+    /// An adjustment made to the [CIE 1931] chromaticity *x* value.
+    ///
+    /// Positive values make the colors redder. Negative values make the colors
+    /// bluer. This has no effect on luminance (brightness).
+    ///
+    /// [CIE 1931]: https://en.wikipedia.org/wiki/CIE_1931_color_space#CIE_xy_chromaticity_diagram_and_the_CIE_xyY_color_space
+    pub temperature: f32,
 
-    /// Saturation adjustment applied before tonemapping.
-    /// Values below 1.0 desaturate, with a value of 0.0 resulting in a grayscale image
-    /// with luminance defined by ITU-R BT.709.
-    /// Values above 1.0 increase saturation.
-    pub pre_saturation: f32,
+    /// An adjustment made to the [CIE 1931] chromaticity *y* value.
+    ///
+    /// Positive values make the colors more magenta. Negative values make the
+    /// colors greener. This has no effect on luminance (brightness).
+    ///
+    /// [CIE 1931]: https://en.wikipedia.org/wiki/CIE_1931_color_space#CIE_xy_chromaticity_diagram_and_the_CIE_xyY_color_space
+    pub tint: f32,
+
+    /// An adjustment to the [hue], in radians.
+    ///
+    /// Adjusting this value changes the perceived colors in the image: red to
+    /// yellow to green to blue, etc. It has no effect on the saturation or
+    /// brightness of the colors.
+    ///
+    /// [hue]: https://en.wikipedia.org/wiki/HSL_and_HSV#Formal_derivation
+    pub hue: f32,
 
     /// Saturation adjustment applied after tonemapping.
     /// Values below 1.0 desaturate, with a value of 0.0 resulting in a grayscale image
     /// with luminance defined by ITU-R BT.709
     /// Values above 1.0 increase saturation.
     pub post_saturation: f32,
+
+    /// The luminance (brightness) ranges that are considered part of the
+    /// "midtones" of the image.
+    ///
+    /// This affects which [`ColorGradingSection`]s apply to which colors. Note
+    /// that the sections smoothly blend into one another, to avoid abrupt
+    /// transitions.
+    ///
+    /// The default value is 0.2 to 0.7.
+    pub midtones_range: Range<f32>,
 }
 
-impl Default for ColorGrading {
+/// The [`ColorGrading`] structure, packed into the most efficient form for the
+/// GPU.
```

Comment:
Out of interest and learning, how/why is this most efficient?
I'm thinking it has to do with the fact that the fields of the struct are of decreasing size?

Reply:
Because:

(1) Shadows, midtones, and highlights are split out into struct-of-vectors instead of array-of-structs, allowing the combination to be done with a dot product;

(2) White balance is baked into a 3x3 matrix ahead of time so the work doesn't have to be redone for every pixel.